torch/lib/c10d/TCPStore.cpp

```diff
@@ -428,17 +455,26 @@ TCPStore::TCPStore(
       if (numWorkers.value_or(-1) >= 0 && waitWorkers) {
         waitForWorkers();
       }
+
+      // socket to handle requests from server
+      listenSocket_ = tcputil::connect(
```

Comment:
Can't we just reuse `storeSocket_` instead of having another socket for the same addr and port?

Reply:
We could do this, but that would involve rewriting all the current methods and some util methods to be asynchronous, since there is currently no way do something like execute a callback while in the middle of `wait()`. To get around this we created a dedicated listen thread which reads from its own socket to handles such asynchronous methods.